### PR TITLE
Offer join form #982

### PR DESCRIPTION
**Pull request by OtisRed**
This component joins offer and if succesful redirects user to offer details.

There is no new imports I think.

No idea how to test it but that could be taken care of in other PR - I think.

### CODE REVIEW

File: frontend/src/app/offers/offer-join-form/offer-join-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import {UserService} from '../../user.service';
+
+@Component({
+  selector: 'volontulo-offer-join-form',
+  templateUrl: './offer-join-form.component.html'
+})
+
+export class OfferJoinFormComponent implements OnInit {
+  public joinForm: FormGroup = this.fb.group({
+    applicant_email: [''],
```
Comment (juliaanholcer) [inline frontend/src/app/offers/offer-join-form/offer-join-form.component.ts]: IMHO camelCase will be better for TypeScript.

File: frontend/src/app/offers/offer-join-form/offer-join-form.component.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <div class="form-group-row">
+      <label class="col-md-3 col-form-label" for="{{ message }}">Możliwe uwagi:</label>
+      <div class="col-md-9">
+        <textarea class="form-control" rows="3" formControlName="message"></textarea>
+        <volontulo-form-error [fc]="joinForm.controls.message" [minLength]="10" [maxLength]="2000"></volontulo-form-error>
+      </div>
+    </div>
+    <div class="form-group row">
+      <div>
+        <button type="submit" class="btn-primary">
```
Comment (juliaanholcer) [inline frontend/src/app/offers/offer-join-form/offer-join-form.component.html]: missed `btn` class

Comment (juliaanholcer) [inline frontend/src/app/offers/offer-join-form/offer-join-form.component.html]: image
This form doesn't look good - smth is wrong with grid and RWD doesn't work as well.

File: frontend/src/app/offers/offer-join-form/offer-join-form.component.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<div class="container">
+  <form [formGroup]="joinForm" (ngSubmit)="onSubmit()">
+    <div class="form-group row">
+      <label class="col-form-label" for="{{ email }}">Email:</label>
+        <div class="col-md-5">
+          <input disabled class="form-control" formControlName="applicant_email" id="{{ email }}"/>
```
Comment (juliaanholcer) [inline frontend/src/app/offers/offer-join-form/offer-join-form.component.html]: You use {{ }} to bind to `email` variable, that is not defined in the component, so as a result, you get empty for attributes in HTML (same problem for all other labels). Define them just using strings, and if you define `for` attribute in a `label` don't forget to define `id' in `input`, so focus can be set after clicking on label.

**Comment by OtisRed** [comment]
Because of git problems I needed to open this change in another PR. Go to #1066

**Pull request closed without merging**